sharding/handler/utils/log_parser.py

```diff
@@ -0,0 +1,70 @@
+from eth_utils import (
+    to_checksum_address,
+    decode_hex,
+    big_endian_to_int,
+)
+
+from sharding.contracts.utils.smc_utils import (
+    get_smc_json,
+)
+from sharding.handler.exceptions import (
+    LogParsingError,
+)
+
+
+class LogParser(object):
```

Comment:
Shouldn't this functionality be part of web3? If it is, it might be better to just reuse it than rewrite it.

Reply:
Web3 has log parser functionality but it's not a stand alone api. It only has api that parse the whole transaction receipt, e.g., `logs = myContract.events.myEvent().processReceipt(tx_receipt)`